2.10 parse error

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**What version of ESLint are you using?**

2.10

**What parser (default, Babel-ESLint, etc.) are you using?**

Tried both default and babel-eslint

**Please show your full configuration:**

```
{
	"extends": "ember",

	"parser": "babel-eslint",

	"ecmaFeatures": {
		modules: true,
		blockBindings: true,
		arrowFunctions: true,
		objectLiteralShorthandMethods: true,
		objectLiteralShorthandProperties: true,
		templateStrings: true
	},

	"rules": {
		"quote-props": [2, "consistent-as-needed"],
		"no-ternary": 0,
		"newline-per-chained-call": 0,
		"arrow-body-style": 0,
		"no-warning-comments": 0,
		"max-len": [1, 200],
		"camelcase": 0,
		"space-before-function-paren": [1, "never"],
		"one-var": [ 1, "never" ],
		"indent": [2, 2, {SwitchCase: 1}],
		"object-shorthand": [ 2, "always" ],
		"prefer-const": 0,
		"comma-dangle": 0,
		"spaced-comment": 1,
		"object-curly-spacing": 0,
		"arrow-spacing": [ 1, {
			before: true,
			after: true
		} ],
		"no-restricted-syntax": 0,

		"no-console": 0,
		"no-magic-numbers": 0,
		"valid-jsdoc": [2, {
			"requireReturn": false
		}]
	},

	"globals": {
		"Ember": true,
		"$": true,
		"Edgar": true,
		"visit": true,
		"andThen": true,
		"find": true,
		"findWithAssert": true,
		"click": true,
		"triggerEvent": true,
		"wait": true,
		"ok": true,
		"equal": true,
		"notEqual": true,
		"deepEqual": true,
		"strictEqual": true,
		"throws": true,
		"fillIn": true,
		"exists": true,
		"currentPath": true,
		"currentURL": true,
		"currentRouteName": true,
		"breadcrumbValidator": true,
		"renderProfiler": true,
		"expect": true,
		"_": true,
		"test": true,
		"module": true,
		"YAMP": true,
		"App": true
	}
}

```

**What did you do? Please include the actual source code causing the issue.**

```js
test('Create New Rule', async function() {
});
```

**What did you expect to happen?**

No parse error

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Workspace/tests/acceptance/rule-test.js
  1:31  error  Parsing error: Unexpected token function
```

--- a/lib/config.js
+++ b/lib/config.js
@@ -243,25 +243,41 @@ Config.prototype.getConfig = function(filePath) {
         config = ConfigOps.merge(config, this.useSpecificConfig);
     }
 
-    // Step 5: Merge in command line settings: environments, globals, parser,
-    // parserOptions, rules
-    debug("Merging command line settings");
-    config = ConfigOps.merge(config, {
-        env: this.env,
-        globals: this.globals,
-        parser: this.parser,
-        parserOptions: this.parserOptions,
-        rules: this.options.rules || {}
-    });
+    // Step 5: Merge in command line environments
+    debug("Merging command line environment settings");
+    config = ConfigOps.merge(config, { env: this.env });
 
-    // Step 6: Merge in command line plugins
+    // Step 6: Merge in command line rules
+    if (this.options.rules) {
+        debug("Merging command line rules");
+        config = ConfigOps.merge(config, { rules: this.options.rules });
+    }
+
+    // Step 7: Merge in command line globals
+    config = ConfigOps.merge(config, { globals: this.globals });
+
+    // Only override parser if it is passed explicitly through the command line or if it's not
+    // defined yet (because the final object will at least have the parser key)
+    if (this.parser || !config.parser) {
+        config = ConfigOps.merge(config, {
+            parser: this.parser
+        });
+    }
+
+    if (this.parserOptions) {
+        config = ConfigOps.merge(config, {
+            parserOptions: this.parserOptions
+        });
+    }
+
+    // Step 8: Merge in command line plugins
     if (this.options.plugins) {
         debug("Merging command line plugins");
         Plugins.loadAll(this.options.plugins);
         config = ConfigOps.merge(config, { plugins: this.options.plugins });
     }
 
-    // Step 7: Apply environments to the config if present
+    // Step 9: Apply environments to the config if present
     if (config.env) {
         config = ConfigOps.applyEnvironments(config);
     }

--- a/lib/options.js
+++ b/lib/options.js
@@ -57,7 +57,6 @@ module.exports = optionator({
         {
             option: "parser",
             type: "String",
-            default: "espree",
             description: "Specify the parser to be used"
         },
         {

--- a/tests/lib/options.js
+++ b/tests/lib/options.js
@@ -356,12 +356,6 @@ describe("options", function() {
 
             assert.equal(currentOptions.parser, "test");
         });
-
-        it("should return a espree if --parser is not passed", function() {
-            var currentOptions = options.parse("");
-
-            assert.equal(currentOptions.parser, "espree");
-        });
     });
 
     describe("--print-config", function() {

